src/Transport/DockerComposeTransport.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Consolidation\SiteProcess\Transport;
+
+use Drush\Drush;
+use Psr\Log\LoggerInterface;
+use Robo\Common\IO;
+use Symfony\Component\Console\Style\OutputStyle;
+use Symfony\Component\Process\Process;
+use Consolidation\SiteProcess\Util\RealtimeOutputHandler;
+use Consolidation\SiteProcess\Util\Escape;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Consolidation\SiteAlias\AliasRecord;
+use Consolidation\SiteProcess\Util\Shell;
+
+/**
+ * DockerComposeTransport knows how to wrap a command such that it executes on a Docker Compose service.
+ */
+class DockerComposeTransport implements TransportInterface
+{
+    protected $tty;
+    protected $siteAlias;
+    protected $cd;
+
+    public function __construct(AliasRecord $siteAlias)
+    {
+        $this->siteAlias = $siteAlias;
+    }
+
+    /**
+     * inheritdoc
+     */
+    public function configure(Process $process)
+    {
+        $this->tty = $process->isTty();
+    }
+
+    /**
+     * inheritdoc
+     */
+    public function wrap($args)
+    {
+        $transport = ['docker-compose', 'exec'];
+        $transportOptions = $this->getTransportOptions();
+        $commandToExecute = $this->getCommandToExecute($args);
+
+        return array_merge(
+            $transport,
+            $transportOptions,
+            $commandToExecute
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function addChdir($cd, $args)
+    {
+        $this->cd = $cd;
+    }
+
+    /**
+     * getTransportOptions returns the transport options for the tranport
+     * mechanism itself
+     */
+    protected function getTransportOptions()
+    {
+        $transportOptions[] = $this->siteAlias->get('service');
+        if ($options = $this->siteAlias->get('exec.options')) {
+            array_unshift($transportOptions, Shell::preEscaped($options));
+        }
+        // Adding -T breaks sql:cli and php:cli so we never do it for now.
```

Comment:
-T is "no tty"; maybe this just needed to be `!$this->tty`?  Seems there should be some situations where it's helpful to not have the tty.